### ISSUE TITLE
Ansible-lint does not catch invalid variable names

### ISSUE THREAD

**Issue by pingiun** [ansible-community/ansible-lint#447]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
ansible 2.7.5
  config file = None
  configured module search path = [u'/Users/jelle/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
ansible-lint --version
ansible-lint 4.0.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

Invalid variable names, e.g. names that contain the `-` character or start with numbers, should be detected and warned about.

[The ansible documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#creating-valid-variable-names) shows valid variable and invalid names.

# Actual Behaviour

Testing playbook:

```yml
- name: "Testing ansible-lint with invalid variable names"
  hosts: 127.0.0.1
  connection: local

  tasks:
    - name: "Test dash in name"
      command: "echo {{ invalid-name }}"
      vars:
        invalid-name: "test"
    - name: "Test digit at the start"
      command: "echo {{ 4test }}"
      vars:
        4test: "test"
    - name: "Test reserved name"
      command: "echo {{ append }}"
      vars:
        append: "test"
```

`ansible-lint` just gives the following output:

```
[301] Commands should not change things if nothing needs doing
test.yml:14
Task/Handler: Test reserved name
```

Ansible itself can give error messages for the first two invalid variables:

```
ERROR! Invalid variable name in vars specified for Task: 'invalid-name' is not a valid variable name

The error appears to have been in '/Users/jelle/Sync/Projects/serverconfig/test.yml': line 9, column 9, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

      vars:
        invalid-name: "test"
        ^ here
```
and
```
ERROR! Invalid variable name in vars specified for Task: '4test' is not a valid variable name

The error appears to have been in '/Users/jelle/Sync/Projects/serverconfig/test.yml': line 13, column 9, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

      vars:
        4test: "test"
        ^ here
```

But `ansible-playbook` gives no error for the last test (reserved name) so it's even more important that `ansible-lint` catches it.

**Comment by awcrosby** [comment]
This sounds like a good addition. This probably should be implemented after the completion of #354 so vars files can also be checked.

**Comment by miyamadoKL** [comment]
Hello.
Ansible gives the error like below when I use loop with a local variable named "groups", also defined as a magic variable and call "{{ item.foo }}" in the looped task.

```
"msg": "The conditional check 'target in item.team' failed. 
The error was: error while evaluating conditional (target in item.team): Unable to look up a name or access an attribute in template string ({% if target in item.team %} True {% else %} False {% endif %}).
Make sure your variable name does not contain invalid characters like '-': argument of type 'StrictUndefined' is not iterable

The error appears to have been in '/var/lib/awx/projects/_20__ansible_repo/roles/os_users/tasks/group.yml': line 2, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

---
- name: Ensure group exists
  ^ here",
```

The task file and the variable I used are below.
- roles/os_users/tasks/group.yml
```
---
- name: Ensure group exists
  group:
    name: "{{ item.name }}"
    gid: "{{ item.gid }}"
    state: "{{ item.state }}"
  loop: "{{ groups }}"
  when: target in item.team
```

- vars.yml
```
---
groups:
  - name: somegroup
    state: present
    team:
      - someteam
  - name: anothergroup
    state: present
    team:
      - anotherteam
```
I give the variable "target" as an extra variable by the AWX template and try to add only "somegroup".
- extra variables
```
---
target: somegroup
```

I solved this problem by changing the variable's name from "groups" to the another one, but report as an example that should be catched by Ansible-lint.
